yt_dlp/extractor/duoplay.py

```diff
@@ -0,0 +1,107 @@
+from .common import InfoExtractor
+from ..utils import (
+    extract_attributes,
+    get_element_text_and_html_by_tag,
+    int_or_none,
+    join_nonempty,
+    str_or_none,
+    try_call,
+    unified_timestamp,
+)
+from ..utils.traversal import traverse_obj
+
+
+class DuoplayIE(InfoExtractor):
+    _VALID_URL = r'https://duoplay\.ee/(?P<id>\d+)/[\w-]+/?(?:\?(?:[^#]+&)?ep=(?P<ep>\d+))?'
+    _TESTS = [{
+        'note': 'Siberi võmm S02E12',
+        'url': 'https://duoplay.ee/4312/siberi-vomm?ep=24',
+        'md5': '1ff59d535310ac9c5cf5f287d8f91b2d',
+        'info_dict': {
+            'id': '4312_24',
+            'ext': 'mp4',
+            'title': 'Operatsioon "Öö"',
+            'thumbnail': r're:https://.+\.jpg(?:\?c=\d+)?$',
+            'description': 'md5:8ef98f38569d6b8b78f3d350ccc6ade8',
+            'upload_date': '20170523',
+            'timestamp': 1495567800,
+            'series': 'Siberi võmm',
+            'series_id': '4312',
+            'season': 'Season 2',
+            'season_number': 2,
+            'episode': 'Operatsioon "Öö"',
+            'episode_number': 12,
+            'episode_id': 24,
+        },
+    }, {
+        'note': 'Empty title',
+        'url': 'https://duoplay.ee/17/uhikarotid?ep=14',
+        'md5': '6aca68be71112314738dd17cced7f8bf',
+        'info_dict': {
+            'id': '17_14',
+            'ext': 'mp4',
+            'title': 'Ühikarotid',
+            'thumbnail': r're:https://.+\.jpg(?:\?c=\d+)?$',
+            'description': 'md5:4719b418e058c209def41d48b601276e',
+            'upload_date': '20100916',
+            'timestamp': 1284661800,
+            'series': 'Ühikarotid',
+            'series_id': '17',
+            'season': 'Season 2',
+            'season_number': 2,
+            'episode': None,
+            'episode_number': None,
+            'episode_id': 14,
```

Comment:
Oh, one last thing no need to include `None`. We also need to add the release year
```suggestion
            'episode_id': 14,
            'release_year': 2010,
```